internal/configs/virtualserver.go

```diff
@@ -2098,6 +2106,14 @@ func (vsc *virtualServerConfigurator) generateSSLConfig(owner runtime.Object, tl
 	return &ssl
 }
 
+func (vsc *virtualServerConfigurator) generateDos(vsr runtime.Object, resource *appProtectDosResource) *version2.Dos {
+	if !vsc.isDosEnabled && resource != nil {
```

Comment:
if we start an Ingress Controller without enabled DOS, and have a VSR with something like this:
```yaml
  subroutes:
  - action:
      pass: coffee
    dos: dos-resource-2
    path: /coffee
```
`generateDos` will be called with `resource` nil, because in the Controller the following code:
```
		if r.Dos != "" {
			routeDosEx, err := lbc.dosConfiguration.GetValidDosEx(virtualServer.Namespace, r.Dos
```
will return nil.

This means the warning reporting code will never be called, and users will not see any warnings.

Reply:
I've updated the PR with a check during validation to see if dos is enabled.

Reply:
After the validation was added in the validation package, I can't think of a case when this check `!vsc.isDosEnabled && resource != nil` will evaluated to `true`. If it is dead code, could you remove it?

Reply:
removed